### PR TITLE
Add crystal/source-gpu-build.yaml

### PR DESCRIPTION
**Pull request by j-rivero**
Added new file to manage the gpu jobs. The diff with the standard `source-build.yaml` file is:

```diff
--- source-build.yaml	2019-04-26 15:44:48.800831704 +0000
+++ source-gpu-build.yaml	2019-04-26 15:44:49.396847412 +0000
@@ -5,6 +5,7 @@
   RTI_NC_LICENSE_ACCEPTED: 'yes'
 build_tool: colcon
 jenkins_commit_job_priority: 57
+jenkins_job_label: gpubuildagent
 jenkins_job_timeout: 120
 jenkins_pull_request_job_priority: 47
 notifications:
@@ -81,6 +82,9 @@
   urls:
   - http://repo.ros2.org/ubuntu/testing
   - http://repositories.ros.org/ubuntu/main
+repository_whitelist:
+- gazebo_ros_pkgs
+skip_ignored_repositories: true
 targets:
   ubuntu:
     bionic:
```

Open questions for me:
 * How to test these changes?
 * Do we need to implement the same for release-* ci-* yaml files?

### CODE REVIEW

**Comment by nuclearsandwich** [comment]
>     * How to test these changes?

We'll start by deploying the GPU jobs to a test farm with an attached GPU agent and make sure they can run there.

>     * Do we need to implement the same for release-* ci-* yaml files?

I don't think it's feasible (test run changes would require changes to the debian metadata generated by bloom) or cost-effective to run GPU tests in the release jobs.

I don't know that there's any reason we couldn't run GPU tests on CI jobs but I also don't see an immediate reason to do so. Especially if we're trying to keep the number of GPU agents to a minimum.

File: index.yaml [hunk 1]
```diff
@@ -8,13 +8,13 @@ distributions:
...
     notification_emails:
-      - steven+build.ros2.org@openrobotics.org
-      - ros2-buildfarm-crystal@googlegroups.com
+      - jrivero+build.ros2.org@openrobotics.org
     release_builds:
       default: crystal/release-build.yaml
       ubv8: crystal/release-bionic-arm64-build.yaml
     source_builds:
       default: crystal/source-build.yaml
+      gpu: crystal/source-gpu-build.yaml
```
Comment (dirk-thomas) [inline index.yaml]: Since Crystal is effectively EOL please remove Crystal specific changes from this PR.

**Comment by dirk-thomas** [comment]
@j-rivero What is the status of this PR?

File: index.yaml (hunk 1 above)
Comment (j-rivero) [reply index.yaml]: 03c39ee

**Comment by j-rivero** [comment]
> @j-rivero What is the status of this PR?

totally forgotten sorry. I've update it to latest master and include foxy.

**Comment by dirk-thomas** [comment]
> I've update it to latest master and include foxy.

The PR still has conflicts. Also the title should probably be updated.

**Comment by j-rivero** [comment]
outdated

**Pull request closed without merging**